### ISSUE TITLE
Trying to find out how to modify resolution

### ISSUE THREAD

**Issue by HitlerWasJew** [mgsweet/Apex-NoRecoil-2021#12]
So i get the point of targeting specific pixels and getting their color i tried modifying to my resolution but i can't find were to point so the script can identify the weapon in use. could you make a tut on how to ? all i need is to find out were to look for pixels.
I tried on 3 random spots on the weapon but i can't make it work.

Looks like this 

; x, y pos for weapon1 and weapon 2
global WEAPON_1_PIXELS = [1300, 868]
global WEAPON_2_PIXELS = [1437, 869]
; weapon color
global LIGHT_WEAPON_COLOR = 0x2D547D
global HEAVY_WEAPON_COLOR = 0x596B38
global ENERGY_WEAPON_COLOR = 0x286E5A
global SUPPY_DROP_COLOR = 0x3701B2

; three x, y check point, true means 0xFFFFFFFF
; light weapon
global R99_PIXELS := [1319, 815, true, 1371, 817, false, 1368, 830, true]
global R301_PIXELS := [1367, 823, false, 1365, 813, true, 1380, 809, true]

Should i pick 3 random pixels on each gun image? i tried taking screen shots in game and then i pulled  3 spots on the weapon icon while in full screen. ( My res is 1600x900 )

**Comment by mgsweet** [comment]
I will take a video this weekend. But my detection way is not as smart as the [python one](https://github.com/mgsweet/Apex-NoRecoil-2021/tree/main/python). Take a look at @krampus-nuggets 's work in [/python/main.py](https://github.com/mgsweet/Apex-NoRecoil-2021/blob/main/python/main.py):
```
weapon_one_coordinates = {
    "left": 1555,
    "top": 1030,
    "width": 110,
    "height": 25
}

weapon_two_coordinates = {
    "left": 1715,
    "top": 1030,
    "width": 110,
    "height": 25
}
```
Changing this to your resolution setting may need less work.

**Comment by hitch3ns** [comment]
sorry for being ignorant is there tutorial on how to use python script ? does it make different from using current AHK as is, speed wise, detection wise ?

**Comment by VStorm001** [comment]
> sorry for being ignorant is there tutorial on how to use python script ? does it make different from using current AHK as is, speed wise, detection wise ?

Ahk must press the hotkey to let it know what to do at the moment.

**Comment by krampus-nuggets** [comment]
As noted by @mgsweet above, the values supplied in those dictionaries basically outline the dimensions and positions of the screenshot that is taken by the script of weapon names in slot 1 and 2. If you re-position and re-size the rectangles to match your display resolution the program should work as expected.

**TL;DR**

Re-position and Re-size the screenshot area correctly over the weapon slots and the program should work as expected.

Additional Info:

* I use the following tool to get the correct dimensions - [link](http://www.pegtop.net/pmeter/)
* An example of how to gather the data is in the repository here - [python/images/info.txt](https://github.com/mgsweet/Apex-NoRecoil-2021/blob/main/python/images/info.txt)

cc @hitch3ns

**Comment by HitlerWasJew** [comment]
> As noted by @mgsweet above, the values supplied in those dictionaries basically outline the dimensions and positions of the screenshot that is taken by the script of weapon names in slot 1 and 2. If you re-position and re-size the rectangles to match your display resolution the program should work as expected.
> 
> **TL;DR**
> 
> Re-position and Re-size the screenshot area correctly over the weapon slots and the program should work as expected.
> 
> Additional Info:
> 
> * I use the following tool to get the correct dimensions - [link](http://www.pegtop.net/pmeter/)
> * An example of how to gather the data is in the repository here - [python/images/info.txt](https://github.com/mgsweet/Apex-NoRecoil-2021/blob/main/python/images/info.txt)
> 
> cc @hitch3ns

Thanks for the info i will try this and post results also i will try to implement more recoil patterns with pattern generator https://github.com/CharlesDankoff/ultra_secret_scripts

**Comment by HitlerWasJew** [comment]
i can understand were to mark the weapon name box? 

https://prnt.sc/1qajfdm like this?

**Comment by HitlerWasJew** [comment]
I made it work but patterns are not working for me its too fast. Is there a way to change the speed?

**Comment by krampus-nuggets** [comment]
That's weird. I'm guessing it could be because the recoil-patterns are linked to the 1080p screen resolution. Since 1080p contains more pixels than 1600x900 the recoil patterns travel over more distance than required on your monitor. When or If we support other screen resolutions we'd either have to create different sets of recoil-patterns for each resolution or build logic for adjusting the recoil patterns based on the screen resolution. Interesting...

In the meantime, you'd have to create and update the recoil-patterns as well yourself since I don't own a 1600x900 monitor.

**Comment by mgsweet** [comment]
@krampus-nuggets 
I think u need to provide a way for setting the mouse sensitivity.
import a configparser 
and do sth like: 
```
modifier = 2.50/ config['mouse']['sens']
win32api.mouse_event(0x0001, int(recoil_patterns[active_weapon][i][0] * modifier), int(recoil_patterns[active_weapon][i][1] * modifier))
```

**Comment by mgsweet** [comment]
@VStorm001 I think the python one can only support English, since it uses OCR to detect the name of the weapon and uses a mapping from name to pattern. If the name of the weapon is in other language, like Chinese, it would not work.

**Comment by VStorm001** [comment]
> @VStorm001 I think the python one can only support English, since it uses OCR to detect the name of the weapon and uses a mapping from name to pattern. If the name of the weapon is in other language, like Chinese, it would not work.

Can OCR detect this?
https://github.com/zkm00323/ApexPressGod_Run/tree/main/Res/GunImage

**Comment by mgsweet** [comment]
> Can OCR detect this?
> https://github.com/zkm00323/ApexPressGod_Run/tree/main/Res/GunImage
I doubt it. OCR is mainly for characters.

**Comment by krampus-nuggets** [comment]
A task has been set for handling different environments for the python-build, you can track the functionality progress on the project board:

* [External Config](https://github.com/mgsweet/Apex-NoRecoil-2021/projects/1#card-67243307)

In regards to supporting different languages, technically you could install or train a version of tesseract on a Chinese language dataset. Then all you'd have to do is change the keys in the recoil-pattern dictionary to their Chinese counter-parts.

If more users are interested in supporting multiple languages I could potentially look into implementing this.

cc @HitlerWasJew @VStorm001

**Comment by krampus-nuggets** [comment]
@HitlerWasJew - The latest Release should resolve your issue with the python-build. It's an executable with a config file for editing the scan boxes and the sensitivity of the recoil-patterns. You can find it here under assets- https://github.com/mgsweet/Apex-NoRecoil-2021/releases

I will now close this issue, if you have any additional issues with the binary itself please open another issue.
